include/fastdds-statistics-backend/StatisticsBackend.hpp

```diff
@@ -165,7 +167,28 @@ class FASTDDS_STATISTICS_BACKEND_DllAPI StatisticsBackend
      * @param entity_id The ID of the entity to which the resulting entities are related
      * @return All entities of type \c entity_type that are related to \c entity_id
      */
-    static std::vector<EntityId> get_entities(
+    static const std::vector<std::shared_ptr<const database::Entity>> get_entities(
```

Comment:
We should also complete this documentation stating that this method could throw and in which cases.